cicd/_common_deploy_logic.sh

```diff
@@ -112,17 +123,19 @@ function transform_arg {
     # transform components to "$1" options for bonfire
     options=""
     option="$1"; shift;
-    components="$@"
+    components="$*"
     for c in $components; do
         options="$options $option $c"
     done
     echo "$options"
 }
 
-if [ ! -z "$COMPONENTS" ]; then
-    export COMPONENTS_ARG=$(transform_arg --component $COMPONENTS)
+if [ -n "$COMPONENTS" ]; then
+    COMPONENTS_ARG=$(transform_arg --component $COMPONENTS)
+    export COMPONENTS_ARG
 fi
 
-if [ ! -z "$COMPONENTS_W_RESOURCES" ]; then
-    export COMPONENTS_RESOURCES_ARG=$(transform_arg --no-remove-resources $COMPONENTS_W_RESOURCES)
+if [ -n "$COMPONENTS_W_RESOURCES" ]; then
+    COMPONENTS_RESOURCES_ARG=$(transform_arg --no-remove-resources $COMPONENTS_W_RESOURCES)
+    export COMPONENTS_RESOURCES_ARG
```

Comment:
I see this pattern all over, we should likely disable it. It's really quite ugly imo.

Reply:
you have any suggestions on how you envision this should look like ? I understand why it is coded this way, essentially you want a list of args passed in to a command, so in order not to duplicate the `--no-remove-resources` bit that's the only way ... LMK Please

Reply:
What I mean was it sets the variable and then exports it, whereas that happened on just one line previously. It's kind of like how our current stuff breaks our imports into single lines with our python projects.

Reply:
yeah well doing that in one line is actually violating https://www.shellcheck.net/wiki/SC2155

Reply:
yeah I get their reasoning, it's a style thing for me, maybe it's not as ugly to others

Reply:
yeah , the problem we're trying to avoid is to miss the expression evaluation result from the assignation action, which is always exit code 0 (that's what SC2155 is trying to help with), regardless of the syntax preventing us from getting away with a one-liner :)